lib/http_aws_sigv4.c

```diff
@@ -154,6 +154,66 @@ static int compare_header_names(const char *a, const char *b)
   return cmp;
 }
 
+/* merge duplicate header definitions by comma delimiting their values
+   in the order defined the headers are defined */
+static CURLcode merge_duplicate_headers(struct curl_slist *head)
+{
+  struct curl_slist *curr = head;
+
+  while(curr) {
+    struct curl_slist *next = curr->next;
+    if(!next) {
+      break;
+    }
+
+    if(compare_header_names(curr->data, next->data) == 0) {
+      char *colon_next;
+      char *merged;
+      char *val_next;
+      size_t curr_len;
+      size_t val_next_len;
+      size_t new_len;
+
+      colon_next = strchr(next->data, ':');
+
+      DEBUGASSERT(colon_next);
+
+      curr_len = strlen(curr->data);
+      val_next = colon_next + 1;
```

Comment:
If `colon_next` is NULL here, this goes wrong.

Reply:
I can't find a way that we'd get a header without a colon in here. By this point, all the headers should have a `:`.

I figured `DEBUGASSERT` would be enough to ensure this invariant during dev builds. However, if you'd prefer that it be changed to something like 
```
if(!colon_next)
    return /* some CURLcode */
```

I have no issues with that. Just lmk if you'd prefer that and which error code is best here (maybe `CURLE_BAD_FUNCTION_ARGUMENT`?)